elasticsearch_dsl/search_base.py

```diff
@@ -632,24 +722,27 @@ def ensure_strings(fields):
                 return str(fields)
 
         if fields is not None:
-            s._source = fields if isinstance(fields, bool) else ensure_strings(fields)
+            s._source = fields if isinstance(fields, bool) else ensure_strings(fields)  # type: ignore[assignment]
             return s
 
         if kwargs and not isinstance(s._source, dict):
             s._source = {}
 
-        for key, value in kwargs.items():
-            if value is None:
-                try:
-                    del s._source[key]
-                except KeyError:
-                    pass
-            else:
-                s._source[key] = ensure_strings(value)
+        if isinstance(s._source, dict):
```

Comment:
Should kwargs be typed as some kind of mapping to avoid the runtime isinstance call? `kwargs` is always going to be a dict.

Reply:
OK, that's annoying. Thanks.